angr/engines/pcode/lifter.py

```diff
@@ -0,0 +1,1488 @@
+
+    :ivar arch:             The architecture this block is lifted under
+    :vartype arch:          :class:`archinfo.Arch`
+    :ivar statements:       The statements in this block
+    :vartype statements:    list of :class:`IRStmt`
+    :ivar next:             The expression for the default exit target of this block
+    :vartype next:          :class:`IRExpr`
+    :ivar int offsIP:       The offset of the instruction pointer in the VEX guest state
+    :ivar int stmts_used:   The number of statements in this IRSB
+    :ivar str jumpkind:     The type of this block's default jump (call, boring, syscall, etc) as a VEX enum string
+    :ivar bool direct_next: Whether this block ends with a direct (not indirect) jump or branch
+    :ivar int size:         The size of this block in bytes
+    :ivar int addr:         The address of this basic block, i.e. the address in the first IMark
+    """
+
+    __slots__ = (
+        "addr",
+        "arch",
+        "_statements",
+        "next",
+        "_tyenv",
+        "jumpkind",
+        "_direct_next",
+        "_size",
+        "_instructions",
+        "_exit_statements",
+        "default_exit_target",
+        "_instruction_addresses",
+        "data_refs",
+        "behaviors"
+    )
+
+    # The following constants shall match the defs in pyvex.h
+    MAX_EXITS = 400
+    MAX_DATA_REFS = 2000
+
+    def __init__(
+        self,
+        data,
+        mem_addr,
+        arch,
+        max_inst=None,
+        max_bytes=None,
+        bytes_offset=0,
+        traceflags=0,
+        opt_level=1,
+        num_inst=None,
+        num_bytes=None,
+        strict_block_end=False,
+        skip_stmts=False,
+        collect_data_refs=False,
+    ):
+        """
+        :param data:                The bytes to lift. Can be either a string of bytes or a cffi buffer object.
+                                    You may also pass None to initialize an empty IRSB.
+        :type data:                 str or bytes or cffi.FFI.CData or None
+        :param int mem_addr:        The address to lift the data at.
+        :param arch:                The architecture to lift the data as.
+        :type arch:                 :class:`archinfo.Arch`
+        :param max_inst:            The maximum number of instructions to lift. (See note below)
+        :param max_bytes:           The maximum number of bytes to use.
+        :param num_inst:            Replaces max_inst if max_inst is None. If set to None as well, no instruction limit is used.
+        :param num_bytes:           Replaces max_bytes if max_bytes is None. If set to None as well, no  byte limit is used.
+        :param bytes_offset:        The offset into `data` to start lifting at. Note that for ARM THUMB mode, both
+                                    `mem_addr` and `bytes_offset` must be odd (typically `bytes_offset` is set to 1).
+        :param traceflags:          The libVEX traceflags, controlling VEX debug prints.
```

Comment:
Currently `traceflags` and `opt_level` are not used, just here for compatibility. Update comments to indicate this.

Reply:
Fixed